### PR TITLE
[MINI-6076] Primary and secondary color

### PR DESCRIPTION
**Pull request by GondiTilak**
# Description
- Updated JavaScript bridge submodule for iOS SDK to latest commit with support for getHostAppThemeColors feature.
- Added new feature in SDK `getHostAppThemeColors(completionHandler:)` which can be used to send the Host app themes primary and secondary colors to mini app through JS-Bridge.
- Implemented the MessageInterfaceDelegate interface method `getHostAppThemeColors(completionHandler:)` in Sample app to pass back the host apps primary and secondary colors.
- Updated the CHANGELOG.md
- Updated the USERGUIDE.md

## Links
[MINI-6076](https://jira.rakuten-it.com/jira/browse/MINI-6076)
Screenshot 2023-05-08 at 14 20 33

# Checklist
- [x] I have read the [contributing guidelines](CONTRIBUTING.md).
- [x] I have completed the [SDK Development Learning Path](CONTRIBUTING.md#SDK-Development-Learning-Path) (if submitting a feature PR)
- [ ] I wrote/updated tests for new/changed code
- [ ] I removed all sensitive data before every commit, including API endpoints and keys
- [ ] I ran `fastlane ci` without errors

### CODE REVIEW

File: Example/Controllers/SwiftUI/Helper/MiniAppViewMessageDelegator.swift [hunk 1]
```diff
@@ -183,6 +183,16 @@ class MiniAppViewMessageDelegator: NSObject, MiniAppMessageDelegate {
...
+
+    func getHostAppThemeColors(completionHandler: @escaping (Result<HostAppThemeColors?, MASDKError>) -> Void) {
+        completionHandler(
+            .success(
+                HostAppThemeColors(
+                    primaryColor: "#FF008C",
+                    secondaryColor: "#7D64BE")
+            )
+        )
+    }
```
Comment (rleojoseph) [inline Example/Controllers/SwiftUI/Helper/MiniAppViewMessageDelegator.swift:195]: We shouldn't be doing any Hardcoding for this scenario. We should fetch the values from Settings screen may be. Will create a ticket on the same.

File: Sources/Classes/core/JavascriptBridge/MiniAppScriptMessageHandler.swift [hunk 2]
```diff
@@ -825,6 +827,24 @@ extension MiniAppScriptMessageHandler {
...
 }
 
+//MARK: MiniApp GetHostAppThemeColors
+extension MiniAppScriptMessageHandler {
+    func getHostAppThemeColors(with callbackId: String) {
+        hostAppMessageDelegate?.getHostAppThemeColors(completionHandler: { (result) in
+            switch result {
+            case .success(let hostAppColors):
+                guard let hostAppColors = ResponseEncoder.encode(data: hostAppColors) else {
+                    self.executeJavaScriptCallback(responseStatus: .onError, messageId: callbackId, response: getMiniAppErrorMessage(MiniAppErrorType.hostAppError))
```
Comment (rleojoseph) [inline Sources/Classes/core/JavascriptBridge/MiniAppScriptMessageHandler.swift:837]: getMiniAppErrorMessage shouldn't be used. Please use `prepareMAJavascriptError` instead

File: Sources/Classes/core/JavascriptBridge/MiniAppScriptMessageHandler.swift [hunk 3]
```diff
@@ -825,6 +827,24 @@ extension MiniAppScriptMessageHandler {
...
+                    return
+                }
+                self.executeJavaScriptCallback(responseStatus: .onSuccess, messageId: callbackId, response: hostAppColors)
+            case .failure(let error):
+                self.handleMASDKError(error: error, callbackId: callbackId)
```
Comment (rleojoseph) [inline Sources/Classes/core/JavascriptBridge/MiniAppScriptMessageHandler.swift:842]: `handleMASDKError` shouldn't be used. Please use `prepareMAJavascriptError` instead

**Review by rleojoseph — CHANGES_REQUESTED** [review]
Overall looks good to me. Need a few changes in ScriptHandler class

File: Sources/Classes/core/JavascriptBridge/MiniAppScriptMessageHandler.swift (hunk 2 above)
Comment (GondiTilak) [reply Sources/Classes/core/JavascriptBridge/MiniAppScriptMessageHandler.swift:837]: l will check an update this.

File: Sources/Classes/core/JavascriptBridge/MiniAppScriptMessageHandler.swift (hunk 3 above)
Comment (GondiTilak) [reply Sources/Classes/core/JavascriptBridge/MiniAppScriptMessageHandler.swift:842]: l will check an update this.

Comment (GondiTilak) [reply Sources/Classes/core/JavascriptBridge/MiniAppScriptMessageHandler.swift]: I have changed to use the recommended apis by updating error parameter to MiniAppJavaScriptError class object.

File: Sources/Classes/core/JavascriptBridge/MiniAppScriptMessageHandler.swift (hunk 2 above)
Comment (GondiTilak) [reply Sources/Classes/core/JavascriptBridge/MiniAppScriptMessageHandler.swift]: I have changed to use the recommended apis by updating error parameter to MiniAppJavaScriptError class object.

File: Example/Controllers/SwiftUI/Helper/MiniAppViewMessageDelegator.swift (hunk 1 above)
Comment (GondiTilak) [reply Example/Controllers/SwiftUI/Helper/MiniAppViewMessageDelegator.swift]: We will change this implementation with the [MINI-6107](https://jira.rakuten-it.com/jira/browse/MINI-6107) Jira ticket for it.

File: Example/Controllers/SwiftUI/Helper/MiniAppViewMessageDelegator.swift [hunk 4]
```diff
@@ -183,6 +183,16 @@ class MiniAppViewMessageDelegator: NSObject, MiniAppMessageDelegate {
     func closeMiniApp(withConfirmation: Bool, completionHandler: @escaping (Result<Bool, MiniAppJavaScriptError>) -> Void) {
         onShoudCloseMiniApp?(withConfirmation)
     }
+
+    func getHostAppThemeColors(completionHandler: @escaping (Result<HostAppThemeColors?, MiniAppJavaScriptError>) -> Void) {
```
Comment (rleojoseph) [inline Example/Controllers/SwiftUI/Helper/MiniAppViewMessageDelegator.swift:187]: `MiniAppJavaScriptError` is used to return any error that is validated by the SDK itself. You should be using MASDKError here.

Comment (GondiTilak) [reply Example/Controllers/SwiftUI/Helper/MiniAppViewMessageDelegator.swift:187]: Screenshot 2023-05-10 at 19 27 52
If I use MASDKError I will get this.

**Pull request merged**